code/game/objects/structures/xeno.dm

```diff
@@ -426,9 +426,42 @@
 			hugger.go_idle(TRUE)
 	addtimer(CALLBACK(src, .proc/Grow), rand(EGG_MIN_GROWTH_TIME, EGG_MAX_GROWTH_TIME))
 
-/obj/effect/alien/egg/Destroy()
-	QDEL_LIST(egg_triggers)
-	return ..()
+	var/list/turf/target_locations = filled_turfs(src, trigger_size, "circle", FALSE)
+	for(var/i in target_locations)
+		RegisterSignal(i, COMSIG_ATOM_ENTERED, .proc/egg_proximity)
+
+
+///Triggers the egg when hostiles get too close
+/obj/effect/alien/egg/proc/egg_proximity(datum/source, atom/hostile)
+	SIGNAL_HANDLER
+
+	if(status != EGG_GROWN || status == (EGG_BURST|EGG_DESTROYED) ) //Don't bother if we're not grown or we're already burst/destroyed
+		return
+
+	if(!COOLDOWN_CHECK(src, egg_proxy_trigger_cooldown)) //Proxy alert triggered too recently; abort
+		return
+
+	if(!isliving(hostile))
+		return
+
+	var/mob/living/living_triggerer = hostile
+	if(living_triggerer.stat == DEAD) //We don't care about the dead
+		return
+
+	if(issamexenohive(hostile))
+		return
+
+	if(hugger_type && iscarbon(living_triggerer)) //If we have a hugger we then care about whether the target can be facehugged.
+		var/mob/living/carbon_triggerer = living_triggerer
+		if(!carbon_triggerer.can_be_facehugged(hugger))
```

Comment:
this doens't need to be a carbon because carbon_triggerer was a living and it dindt fail compile
```suggestion
	if(hugger_type) //If we have a hugger we then care about whether the target can be facehugged.
		if(!living_triggerer.can_be_facehugged(hugger))
```